### ISSUE TITLE
File manager: Better solution to open the context menu of files when using a screen reader?

### ISSUE THREAD

**Issue by domasofan** [owncloud/core#14435]
Hi all,

I have issues opening the context menu of files which i have dropped into my owncloud install.
Opening the context menu of files isn't always easy.
The method of moving the mouse to the currently selected item doesn't always work reliably.

Google Drive has recently added a very accessible way of moving through the interface and accessing the file list and the menus.

Another solution might also be to add a small button to each file to open the menu.

Or does someone have a better solution?
I am also open for new ideas.

greetings,
simon

**Comment by domasofan** [comment]
Hi,
Forgot to add i used ownCloud 8 and tested with the NVVDA and the JAWS screenreader under windows.
Other screen readers might have the same functions.


I also thought that clicking on the file would open the menu where you then can select open or download or share or favorite or what ever you want.
Maybe might be a little bit irritating for sighted people when the menu just opens then?

Greetings,
Simon

**Comment by jancborchardt** [comment]
Hey @domasofan! Sorry for the problems – I worked on improving the accessibility for ownCloud 8 and the file actions were the one big known problem left. :\

The plan is similar to what you mentioned: Move the file actions (except sharing) into a dropdown. See the discussion in issue https://github.com/owncloud/core/issues/12081

Is it ok if I close this one in favor of the existing issue? We also have another similar related one at https://github.com/owncloud/core/issues/12999

@domasofan also, did you encounter any other accessibility issues? I’m happy to fix them all, it’s just difficult to test since I don’t use the tools daily myself. :)

**Comment by domasofan** [comment]
Hi,

That's fine to me.
If i find something i will report to you.
I also could do audio demos if you want.

greetings,
simon

On Mon, 23 Feb 2015 05:52:24 -0800
  Jan-Christoph Borchardt <notifications@github.com> wrote:
> Hey @domasofan! Sorry for the problems – I worked on improving the 
>accessibility for ownCloud 8 and the file actions were the one big 
>known problem left. :\
> 
> The plan is similar to what you mentioned: Move the file actions 
>(except sharing) into a dropdown. See the discussion in issue 
>https://github.com/owncloud/core/issues/12081
> 
> Is it ok if I close this one in favor of the existing issue? We also 
>have another similar related one at 
>https://github.com/owncloud/core/issues/12999
> 
> @domasofan also, did you encounter any other accessibility issues? 
>I’m happy to fix them all, it’s just difficult to test since I don’t 
>use the tools daily myself. :)
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/owncloud/core/issues/14435#issuecomment-75543591

--
Simon Eigeldinger
simon.eigeldinger@vol.at

**Comment by jancborchardt** [comment]
@domasofan thank you very much! What other accessibility issues in ownCloud do you know of except this file actions one?

**Comment by domasofan** [comment]
hi,

what i have found:

File manager:
* when you click  the favorite link you need to refresh the page to 
see which things have been favorited. i am not sure if this is just 
with jaws. haven't tested with NVDA but NVDA is normally better in 
this.
* in the last row of the files table there is a word which says and.
i don't know what it does but when you try to activate it it seems to 
do nothing much.

General Interface:
* when you look at the top of the page where you can expand to see the 
apps there is at the moment the currently active app on the link text.
don't know if this is a good solution or not.
basically what it does it opens and closes the menu to select which 
app you want to see.
google calls this link apps i guess.

greetings,
simon

On Wed, 25 Feb 2015 05:17:21 -0800
  Jan-Christoph Borchardt <notifications@github.com> wrote:
> @domasofan thank you very much! What other accessibility issues in 
>ownCloud do you know of except this file actions one?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/owncloud/core/issues/14435#issuecomment-75957683

--
Simon Eigeldinger
simon.eigeldinger@vol.at

**Comment by domasofan** [comment]
hi,

what i also have seen now that i am at home and using nvda.
also nvda seems not to get the not changing status of the favorite link.
seems this doesn't update automatically. just after a refresh of the 
browser window.


greetings,
simon

Am 25.02.2015 um 14:17 schrieb Jan-Christoph Borchardt:
> @domasofan thank you very much! What other accessibility issues in ownCloud do you know of except this file actions one?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/owncloud/core/issues/14435#issuecomment-75957683
>

-- 
Simon Eigeldinger
Follow me on Twitter: http://www.twitter.com/domasofan/
E-Mail: simon.eigeldinger@vol.at
MSN: simon_eigeldinger@hotmail.com
ICQ: 121823966
Jabber: domasofan@andrelouis.com

---
Diese E-Mail wurde von Avast Antivirus-Software auf Viren geprüft.
http://www.avast.com
